src/compositor/shells/xdg_v6.rs

```diff
@@ -20,13 +20,24 @@ pub struct XdgV6ShellManager;
 impl XdgV6ShellManagerHandler for XdgV6ShellManager {
     fn new_surface(&mut self,
                    compositor: CompositorHandle,
-                   shell_surface: XdgV6ShellSurfaceHandle)
+                   mut shell_surface: XdgV6ShellSurfaceHandle)
                    -> Option<Box<XdgV6ShellHandler>> {
-        with_handles!([(compositor: {compositor})] => {
-            let server: &mut Server = compositor.into();
-            server.views
-                .push(View::new(Shell::XdgV6(shell_surface.into())));
+
+        let is_toplevel = with_handles!([(shell_surface: {&mut shell_surface})] => {
+            match shell_surface.state().unwrap() {
+                TopLevel(_) => true,
+                _ => false
+            }
         }).unwrap();
+
+        if is_toplevel {
```

Comment:
Don't do this, put this code block in the `TopLevel(_) => {...}` block. There's no reason to return `true` `false` here and then immediantly do something when it's true. In the other case just put the `{}` block.

Reply:
I can't figure out how to access the handle and the underlying shell surface at the same time. I think the handle gets moved in the with_handles! macro. I tried this:

```rust
        with_handles!([(shell_surface: {shell_surface_handle}), (compositor: {compositor})] => {
            match shell_surface.state().unwrap() {
                TopLevel(_) => {
                    let server: &mut Server = compositor.into();
                    server.views
                        .push(View::new(Shell::XdgV6(shell_surface_handle.into())));
                }
                _ => ()
            }
        }).unwrap();
```

And get

```
error[E0382]: capture of moved value: `shell_surface_handle`
```